src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoUtils.java

```diff
@@ -38,15 +44,44 @@ public static TableNameOverride getTableNameOverride(final Class<?> dynamoTable,
         if (table == null) {
             throw new IllegalArgumentException("Missing DynamoDBTable annotation for " + dynamoTable.getName());
         }
-        final Environment env = config.getEnvironment();
-        return new TableNameOverride(env.name().toLowerCase() + "-" + config.getUser() + "-" + table.tableName());
+        return new TableNameOverride(getTableNamePrefix(config) + table.tableName());
     }
 
-    public static String getTableName(final Class<?> dynamoTable, final Config config) {
+    /**
+     * Given a class annotated as DynamoDBTable, gets the fully qualified table name.
+     */
+    public static String getFullyQualifiedTableName(final Class<?> dynamoTable, final Config config) {
         return getTableNameOverride(dynamoTable, config).getTableName();
     }
 
     /**
+     * Given a class annotated as DynamoDBTable, gets the fully qualified table name.
+     */
+    public static String getFullyQualifiedTableName(final String simpleTableName, final Config config) {
+        checkNotNull(simpleTableName);
+        checkNotNull(config);
+        return getTableNamePrefix(config) + simpleTableName;
+    }
+
+    /**
+     * Given a class annotated as DynamoDBTable, gets the fully qualified table name.
+     */
+    public static String getSimpleTableName(final String fullyQualifiedTableName, final Config config) {
+        checkNotNull(fullyQualifiedTableName);
+        checkNotNull(config);
+        final String prefix = getTableNamePrefix(config);
+        if (!fullyQualifiedTableName.startsWith(prefix)) {
+            throw new IllegalArgumentException(fullyQualifiedTableName + " is not a fully qualified table name.");
+        }
+        return fullyQualifiedTableName.substring(prefix.length());
+    }
```

Comment:
Would it be worth creating some simple unit tests for getFullyQualifiedTableName() and getSimpleTableName()?